### DISCUSSION TITLE
WixUI_Mondo asking for requirements for InstallDirDlg?

### DISCUSSION THREAD

**Discussion by Biye123** [wixtoolset/issues#7499] [Questions] [answered]
All I have right now regarding UI code is 
```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs"
     xmlns:ui="http://wixtoolset.org/schemas/v4/wxs/ui">
...
<ui:WixUI Id="WixUI_Mondo" />
<UIRef Id="WixUI_ErrorProgressText" />
...
</Wix>
```
But for some reason, when I compile, I'm getting ICE17 errors related to InstallDirDlg.  If I understood correctly, this dialog is not part of the Mondo package (at least from what I can see in the source code).  Am I missing something?

I can bypass these errors by implementing the required button actions, but it feels like I shouldn't need to?
![image](https://github.com/wixtoolset/issues/assets/32176449/0ae95dcb-7c88-4baa-bcd1-8edeb2c09a2b)

Edit: Also, as suspected, the neither the dialog nor the buttons referenced show up in the installation sequence after a successful compile.  So while its not detrimental to the installation process, it does seem like unnecessary code?

Here's what I added to remove the errors
```
    <UI>
      <DialogRef Id="InstallDirDlg"/>
      <Publish
        Dialog="InstallDirDlg"
        Control="Back"
        Event="NewDialog"
        Value="WelcomeDlg"
      />
      <Publish
        Dialog="InstallDirDlg"
        Control="ChangeFolder"
        Event="NewDialog"
        Value="WelcomeDlg"
      />
    </UI>
```

**Comment by barnson** [comment, marked as answer]
https://github.com/wixtoolset/issues/issues/7436

**Reply by Biye123** [reply]
sorry for duplicate, I only searched discussions and not issues
